CONTRIBUTING.md

```diff
@@ -0,0 +1,159 @@
+Table of contents
+
+ - [Introduction](#introduction)
+ - [Devfile](#devfile)
+ - [Contribute to VS Code Kubernetes tooling extension](#)
+ - [Eclipse Che Sidecar container for Kubernetes plugin](#)
+ 
+## Introduction
+
+There are 2 contribution points:
+- VS Code [Kubernetes tooling extension](https://github.com/Azure/vscode-kubernetes-tools)
+- [Eclipse Che Sidecar container](https://github.com/che-dockerfiles/che-sidecar-kubernetes-tooling) for [Kubernetes plugin](https://github.com/eclipse/che-plugin-registry/tree/master/v3/plugins/ms-kubernetes-tools/vscode-kubernetes-tools)
+
+For contribution to Kubernetes tooling VS Code extension, you don't need to rebuild the container image to code, build and test.
+It is possible to create a Che workspace that would setup all the containers to build and run the extension. The workspace will have a dedicated container with [buildah](https://github.com/containers/buildah) tool, which you can use to build the docker image containing the Kubernetes tooling.
+You will find here a [devfile](devfile.yaml) that will help you to setup the environment.
+
+## Devfile
+
+The devfile could be run on any Che instances. But given the memory requirements of about 8 gigabytes, and some limitations when using the buildah, the devfile has been thoroughly tested only on local Che deployment running in minikube.
+The devfile could be launched through a factory or [chectl](https://github.com/che-incubator/chectl) cli. It's also possible to apply the devfile content when creating a workspace in the dashboard.
+
+The newly created workspace will have the following containers:
+
+- theia-ide
+- che-dev
+- buildah-dev
+
+**Theia-Ide** is used as default container where your Che-Theia editor is running. We also use this container to run another instance of Che-Theia to test the Kubernetes extension. To be ensure that memory is enough for running of two instances of Che-Thea with full set of plugins, the memory limitation for this container is set to 3 gigabytes.
+
+**Che-Dev** container is used to build the Kubernetes tooling extension. To run it in **buildah-dev** container, the [eclipse-che-theia-plugin-remote](https://github.com/eclipse/che-theia/tree/master/extensions/eclipse-che-theia-plugin-remote) extension of [che-theia](https://github.com/eclipse/che-theia) is used. Since this extension is not published to [npmjs](https://www.npmjs.com/search?q=%40eclipse-che), we have to build it from sources. Memory limitation for che-dev container is set to 2 gigabytes.
+
+The third, **buildah-dev** is needed to run Kubernetes tooloing extension as a remote plugin and to build sidecar docker image for Eclipse Che Kubernetes plugin. It should be emphasized that the target directory with build images is not persisted, will be lost after restarting the workspace. You need to push your images to [docker.io](docker.io) or [quay.io](quay.io) before leaving the workspace.
+
+After startup and finishing cloning of projects, your will have
```

Comment:
you, not your. Also consider rephrasing the sentence.

Reply:
done